ingest.go

```diff
@@ -1555,7 +1562,7 @@ func (d *DB) ingest(
 		}
 		// The ingestion overlaps with some entry in the flushable queue.
 		if d.FormatMajorVersion() < FormatFlushableIngest ||
-			d.opts.Experimental.DisableIngestAsFlushable() ||
+			d.opts.Experimental.DisableIngestAsFlushable() || !doFlushableIngest ||
```

Comment:
This is slightly inaccurate, as we still want to do a flushable ingest if there's no excise span or shared/external files, even if doFlushableIngest is false. Maybe we need:

```
canIngestFlushable := d.FormatMajorVersion() >= FormatFlushableIngest && (len(d.mu.mem.queue) < d.opts.MemTableStopWritesThreshold) && !d.opts.Experimental.DisableIngestAsFlushable()
hasExciseOrRemoteFiles := len(shared) > 0 || exciseSpan.Valid() || len(external) > 0

if !canIngestFlushable || (hasExciseOrRemoteFiles && !sstContainsExciseTombstone) {
...
}
```

Reply:
Actually, looks like after this change we _can_ do excise+local files, but we still can't flushable ingest shared or external files. So scratch what I wrote above and move the remote files stuff into `canIngestFlushable`.